crypto/asn1/a_int.c

```diff
@@ -229,12 +229,10 @@ static size_t asn1_put_uint64(unsigned char b[sizeof(uint64_t)], uint64_t r)
 }
 
 /*
- * Absolute value of INT64_MIN: we can't just use -INT64_MIN as it produces
+ * Absolute value of INT64_MIN: we can't just use -INT64_MIN as gcc produces
  * overflow warnings.
  */
-
-#define ABS_INT64_MIN \
-    ((uint64_t)INT64_MAX + (uint64_t)(-(INT64_MIN + INT64_MAX)))
+#define ABS_INT64_MIN ((uint64_t)INT64_MAX + (-(INT64_MIN + INT64_MAX)))
```

Comment:
> maybe ABS_INT32_MIN is the one that needs fixing

In such case x_int64.c needs overhaul too. What I'm saying is that INT32_MAX+1 is consistent with the rest of the x_int64.c. INT64_MAX+1 wouldn't be with the rest of a_int.c. I can make suggestion even for x_int64.c...